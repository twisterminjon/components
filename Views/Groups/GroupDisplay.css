.groupdisplay-wrap {
  height: calc(100% - var(--header-height));
}

.groupdisplay-content {
  display: flex;
  flex-direction: column;

  height: 100%;
  width: 100%;
}

.groupdisplay-list {
  flex: 1;
  /* This is the scrolling portion of the page
     it's height is determined by the flexbox but we need to
     account for the fixed footer
  */
  margin-bottom: 86px;

  width: 100%;

  overflow-y: auto;
}

.groupdisplay-text {
  display: block;
  margin-top: 25px;
  font-size: 22px;
  line-height: 31px;
  width: 100%;
  text-align: center;
  padding: 0 20px;
  color: var(--brandcolor);
}
