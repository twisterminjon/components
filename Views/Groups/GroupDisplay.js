import React from 'react'
import PropTypes from 'prop-types'

import GroupHeader from './GroupHeader'
import GroupFooter from './GroupFooter'
import UserList from '../../Organisms/UserList/UserList'
import DocTitle from '../../Atoms/DocTitle/DocTitle'
import SpinnerDots from '../../Atoms/SpinnerDots/SpinnerDots'

import './GroupDisplay.css'

GroupDisplay.propTypes = {
  /** An array of groups to display */
  group: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    users: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        displayName: PropTypes.string.isRequired,
        profileImage: PropTypes.string,
        overallStatus: PropTypes.string.isRequired,
        favorite: PropTypes.bool.isRequired,
      })
    ),
    /** Enterprise for this user */
    enterprise: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }),
  }),

  /** Called when the group breadcrumb is clicked */
  onGoBack: PropTypes.func.isRequired,

  /** Start a call with a group */
  onGroupCall: PropTypes.func.isRequired,

  /** Start a call with a user */
  onCallUser: PropTypes.func.isRequired,

  /** Start a messaging with a user */
  onMessageUser: PropTypes.func,

  /** If true, will show the message button */
  canMessage: PropTypes.bool,

  /** If true, will show the dial button */
  canCall: PropTypes.bool,

  /** Whether or not there is an available group member */
  available: PropTypes.bool.isRequired,

  /** Whether or not the group user list is loading */
  loading: PropTypes.bool.isRequired,

  /** The current group user list page number, starting from 0 */
  pageNumber: PropTypes.number.isRequired,

  /** Called, with the the page number, after the user list page number changes */
  onPageNumberChange: PropTypes.func.isRequired,
}

GroupDisplay.defaultProps = {
  onMessageUser: () => {},
  canMessage: false,
}

export default function GroupDisplay({
  group,
  onGoBack,
  onGroupCall,
  onCallUser,
  onMessageUser,
  canMessage,
  canCall,
  available,
  loading,
  pageNumber,
  onPageNumberChange,
}) {
  const handleCallUser = user => {
    onCallUser(user)
  }

  const handleMessageUser = user => {
    onMessageUser(user)
  }

  return (
    <div className="groupdisplay-wrap">
      <DocTitle title="Group Call" />
      <div className="groupdisplay-content">
        <GroupHeader name={group.name} onClick={onGoBack} />
        {
          // FIXME: Add loading spinner to UserList itself to reduce code duplication
        }
        {loading && (
          <div
            style={{
              position: 'absolute',
              bottom: 60,
              height: 40,
              display: 'inline-flex',
              width: '100%',
              alignContent: 'center',
              justifyContent: 'center',
              zIndex: 99999,
            }}>
            <SpinnerDots />
          </div>
        )}
        <div className="groupdisplay-list">
          <UserList
            canMessage={canMessage}
            canCall={canCall}
            userList={group.users}
            onCall={handleCallUser}
            onMessage={handleMessageUser}
            loading={loading}
            pageNumber={pageNumber}
            onPageNumberChange={onPageNumberChange}
          />
          <div style={{ height: 'var(--bottom-whitespace-v1)' }} />
        </div>
        <GroupFooter onClick={onGroupCall} disabled={!available || !canCall} />
      </div>
    </div>
  )
}
