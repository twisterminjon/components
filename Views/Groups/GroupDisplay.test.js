import React from 'react'
import GroupDisplay from './GroupDisplay'

describe('GroupDisplay', () => {
  const mockFun = jest.fn()
  const userid = '5'
  const group = {
    id: '1',
    name: 'some group',
    users: [
      {
        id: '2',
        displayName: 'first last',
        profileImage: null,
        overallStatus: 'available',
        favorite: false,
      },
      {
        id: '3',
        displayName: 'first ',
        profileImage: 'www.projectified/image.png',
        overallStatus: 'busy',
        favorite: false,
      },
      {
        id: '4',
        displayName: ' last',
        profileImage: null,
        overallStatus: 'offline',
        favorite: false,
      },
    ],
    enterprise: {
      id: '134dfs',
      name: 'some name',
    },
  }

  const groupEmpty = {
    id: '1',
    name: 'some group',
    users: [],
    enterprise: {
      id: '134dfs',
      name: 'some name',
    },
  }

  it('matches the snapshot', () => {
    const wrapper = window.shallow(
      <GroupDisplay
        userid={userid}
        group={group}
        onGoBack={mockFun}
        onGroupCall={mockFun}
        onCallUser={mockFun}
        onFavoriteUser={mockFun}
        enterpriseLogo=""
        canMessage={true}
        canCall={true}
        available={true}
        loading={false}
        pageNumber={0}
        onPageNumberChange={mockFun}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot when not available', () => {
    const wrapper = window.shallow(
      <GroupDisplay
        userid={userid}
        group={group}
        onGoBack={mockFun}
        onGroupCall={mockFun}
        onCallUser={mockFun}
        onFavoriteUser={mockFun}
        enterpriseLogo=""
        canMessage={true}
        canCall={true}
        available={false}
        loading={false}
        pageNumber={0}
        onPageNumberChange={mockFun}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot with no users', () => {
    const wrapper = window.shallow(
      <GroupDisplay
        group={groupEmpty}
        onGoBack={mockFun}
        onGroupCall={mockFun}
        onCallUser={mockFun}
        onFavoriteUser={mockFun}
        canMessage={true}
        available={false}
        loading={false}
        pageNumber={0}
        onPageNumberChange={mockFun}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot when loading', () => {
    const wrapper = window.shallow(
      <GroupDisplay
        userid={userid}
        group={group}
        onGoBack={mockFun}
        onGroupCall={mockFun}
        onCallUser={mockFun}
        onFavoriteUser={mockFun}
        canMessage={true}
        available={true}
        loading={false}
        pageNumber={0}
        onPageNumberChange={mockFun}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
