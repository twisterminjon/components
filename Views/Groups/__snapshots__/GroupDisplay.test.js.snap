// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupDisplay matches the snapshot 1`] = `
<div
  className="groupdisplay-wrap"
>
  <DocTitle
    title="Group Call"
  />
  <div
    className="groupdisplay-content"
  >
    <GroupHeader
      name="some group"
      onClick={[MockFunction]}
    />
    <div
      className="groupdisplay-list"
    >
      <UserList
        canCall={true}
        canMessage={true}
        loading={false}
        onCall={[Function]}
        onMessage={[Function]}
        onPageNumberChange={[MockFunction]}
        pageNumber={0}
        userList={
          Array [
            Object {
              "displayName": "first last",
              "favorite": false,
              "id": "2",
              "overallStatus": "available",
              "profileImage": null,
            },
            Object {
              "displayName": "first ",
              "favorite": false,
              "id": "3",
              "overallStatus": "busy",
              "profileImage": "www.projectified/image.png",
            },
            Object {
              "displayName": " last",
              "favorite": false,
              "id": "4",
              "overallStatus": "offline",
              "profileImage": null,
            },
          ]
        }
      />
      <div
        style={
          Object {
            "height": "var(--bottom-whitespace-v1)",
          }
        }
      />
    </div>
    <GroupFooter
      disabled={false}
      onClick={[MockFunction]}
    />
  </div>
</div>
`;

exports[`GroupDisplay matches the snapshot when loading 1`] = `
<div
  className="groupdisplay-wrap"
>
  <DocTitle
    title="Group Call"
  />
  <div
    className="groupdisplay-content"
  >
    <GroupHeader
      name="some group"
      onClick={[MockFunction]}
    />
    <div
      className="groupdisplay-list"
    >
      <UserList
        canMessage={true}
        loading={false}
        onCall={[Function]}
        onMessage={[Function]}
        onPageNumberChange={[MockFunction]}
        pageNumber={0}
        userList={
          Array [
            Object {
              "displayName": "first last",
              "favorite": false,
              "id": "2",
              "overallStatus": "available",
              "profileImage": null,
            },
            Object {
              "displayName": "first ",
              "favorite": false,
              "id": "3",
              "overallStatus": "busy",
              "profileImage": "www.projectified/image.png",
            },
            Object {
              "displayName": " last",
              "favorite": false,
              "id": "4",
              "overallStatus": "offline",
              "profileImage": null,
            },
          ]
        }
      />
      <div
        style={
          Object {
            "height": "var(--bottom-whitespace-v1)",
          }
        }
      />
    </div>
    <GroupFooter
      disabled={true}
      onClick={[MockFunction]}
    />
  </div>
</div>
`;

exports[`GroupDisplay matches the snapshot when not available 1`] = `
<div
  className="groupdisplay-wrap"
>
  <DocTitle
    title="Group Call"
  />
  <div
    className="groupdisplay-content"
  >
    <GroupHeader
      name="some group"
      onClick={[MockFunction]}
    />
    <div
      className="groupdisplay-list"
    >
      <UserList
        canCall={true}
        canMessage={true}
        loading={false}
        onCall={[Function]}
        onMessage={[Function]}
        onPageNumberChange={[MockFunction]}
        pageNumber={0}
        userList={
          Array [
            Object {
              "displayName": "first last",
              "favorite": false,
              "id": "2",
              "overallStatus": "available",
              "profileImage": null,
            },
            Object {
              "displayName": "first ",
              "favorite": false,
              "id": "3",
              "overallStatus": "busy",
              "profileImage": "www.projectified/image.png",
            },
            Object {
              "displayName": " last",
              "favorite": false,
              "id": "4",
              "overallStatus": "offline",
              "profileImage": null,
            },
          ]
        }
      />
      <div
        style={
          Object {
            "height": "var(--bottom-whitespace-v1)",
          }
        }
      />
    </div>
    <GroupFooter
      disabled={true}
      onClick={[MockFunction]}
    />
  </div>
</div>
`;

exports[`GroupDisplay matches the snapshot with no users 1`] = `
<div
  className="groupdisplay-wrap"
>
  <DocTitle
    title="Group Call"
  />
  <div
    className="groupdisplay-content"
  >
    <GroupHeader
      name="some group"
      onClick={[MockFunction]}
    />
    <div
      className="groupdisplay-list"
    >
      <UserList
        canMessage={true}
        loading={false}
        onCall={[Function]}
        onMessage={[Function]}
        onPageNumberChange={[MockFunction]}
        pageNumber={0}
        userList={Array []}
      />
      <div
        style={
          Object {
            "height": "var(--bottom-whitespace-v1)",
          }
        }
      />
    </div>
    <GroupFooter
      disabled={true}
      onClick={[MockFunction]}
    />
  </div>
</div>
`;
